servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapMergeTest.java

```diff
@@ -96,6 +105,43 @@ public static void afterClass() throws Exception {
         }
     }
 
+    @Test
+    public void mappedRecoverMakesProgress() throws Exception {
+        @SuppressWarnings("unchecked")
+        Subscriber<Integer> mockSubscriber = mock(Subscriber.class);
+        CountDownLatch latchOnSubscribe = new CountDownLatch(1);
+        CountDownLatch latchOnError = new CountDownLatch(1);
+        AtomicReference<Throwable> causeRef = new AtomicReference<>();
+        BlockingQueue<Integer> results = new ArrayBlockingQueue<>(10);
+        doAnswer(a -> {
+            Subscription s = a.getArgument(0);
+            s.request(4);
+            latchOnSubscribe.countDown();
+            return null;
+        }).when(mockSubscriber).onSubscribe(any(Subscription.class));
+        doAnswer(a -> {
+            causeRef.set(a.getArgument(0));
+            latchOnError.countDown();
+            return null;
+        }).when(mockSubscriber).onError(any());
+        doAnswer(a -> {
+            results.add(a.getArgument(0));
+            throw new DeliberateException();
+        }).when(mockSubscriber).onNext(any());
+
+        Processor<Integer, Integer> processor = newPublisherProcessor();
+        toSource(fromSource(processor).flatMapMergeDelayError(i -> from(i + 10).recoverWith(cause ->
+                from(i + 20).concat(failed(cause))))).subscribe(mockSubscriber);
+
+        latchOnSubscribe.await();
+        processor.onNext(1);
+        assertThat(results.take(), is(11));
+        processor.onError(DELIBERATE_EXCEPTION);
+        assertThat(results.take(), is(21));
```

Comment:
The ordering of events in this test is a bit confusing. `results` queue already has `21` before `processor.onError(DELIBERATE_EXCEPTION)` is invoked. Consider moving this check up if this is expected.
Why the concatenated `cause` is not delivered to `mockSubscriber`?

Reply:
oh, bcz we delay error.

Reply:
WDYT about this ordering of events?
```java
        latchOnSubscribe.await();
        processor.onNext(1);
        assertThat(results.take(), is(11));
        assertThat(results.take(), is(21));
        assertThat(causeRef.get(), is(nullValue()));
        processor.onComplete();
        latchOnError.await();
        final Throwable t = causeRef.get();
        assertThat(t, is(instanceOf(CompositeException.class)));
        assertThat(t.getCause(), is(DELIBERATE_EXCEPTION));
```

Reply:
I adjusted slightly but sgtm!